fishtank/src/cluster.ts

```diff
@@ -27,11 +54,55 @@ export class Cluster {
     return this.backend.createNetwork(this.networkName(), { attachable: true, internal: true })
   }
 
-  async spawn(options: { name: string; image?: string }): Promise<void> {
+  async spawn(options: { name: string; image?: string; config?: NodeConfig }): Promise<void> {
     const name = this.containerName(options.name)
-    return this.backend.runDetached(options.image ?? DEFAULT_IMAGE, {
+
+    const args: string[] = []
+    const runOptions: RunOptions = {
       name,
       networks: [this.networkName()],
-    })
+    }
+
+    if (options?.config) {
+      const config = options.config
+
+      if (config.networkId) {
+        args.push(`--networkId=${config.networkId}`)
+      }
+
+      if (config.cliconfig) {
+        if (config.cliconfig.configName) {
+          if (config.cliconfig.dataDir) {
+            args.push(`--config=${config.cliconfig.configName}`)
+          } else {
+            throw new Error('Need to set datadir when config name file is provided.')
+          }
+        }
+
+        if (config.cliconfig.dataDir) {
+          const parsedPath = parse(config.cliconfig.dataDir)
+
+          const dest = join(tmpdir(), name, parsedPath.name)
+          if (!existsSync(dest)) {
+            mkdirSync(dest, {
+              recursive: true,
+            })
+          }
+
+          cpSync(config.cliconfig.dataDir, dest, {
+            recursive: true,
+          })
+
+          args.push(`--datadir=${dest}`)
+          runOptions.volumes = new Map<string, string>([[dest, dest]])
```

Comment:
Use `[dest, '/root/.ironfish']` instead of `[dest, dest]`, so that passing `--datadir=...` becomes unnecessary.

Reply:
Pass to `--datadir=...` is necessary, It’s the most common config user set for the node. it’s supported in all of our interface, eg ironfish cli, existing simulation

It will be useful to spin up node with some existing datadir to simulate certain node state. Eg. my local node is in a state and I want to simulate a network connecting with such node.